# Exploration - Régularité mensuelle TGV

In [1]:
import pandas as pd
pd.set_option("display.max_columns", None)

df = pd.read_csv("../data/raw/regularite_tgv.csv", sep=";")
df.shape

(12544, 26)

In [2]:
df.info()
df.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12544 entries, 0 to 12543
Data columns (total 26 columns):
 #   Column                                Non-Null Count  Dtype  
---  ------                                --------------  -----  
 0   date                                  12544 non-null  object 
 1   service                               12544 non-null  object 
 2   gare_depart                           12544 non-null  object 
 3   gare_arrivee                          12544 non-null  object 
 4   duree_moyenne                         12544 non-null  int64  
 5   nb_train_prevu                        12544 non-null  int64  
 6   nb_annulation                         12544 non-null  int64  
 7   commentaire_annulation                0 non-null      float64
 8   nb_train_depart_retard                12544 non-null  int64  
 9   retard_moyen_depart                   12544 non-null  float64
 10  retard_moyen_tous_trains_depart       12544 non-null  float64
 11  commentaire_ret

,date,service,gare_depart,gare_arrivee,duree_moyenne,nb_train_prevu,nb_annulation,commentaire_annulation,nb_train_depart_retard,retard_moyen_depart,retard_moyen_tous_trains_depart,commentaire_retards_depart,nb_train_retard_arrivee,retard_moyen_arrivee,retard_moyen_tous_trains_arrivee,commentaires_retard_arrivee,nb_train_retard_sup_15,retard_moyen_trains_retard_sup15,nb_train_retard_sup_30,nb_train_retard_sup_60,prct_cause_externe,prct_cause_infra,prct_cause_gestion_trafic,prct_cause_materiel_roulant,prct_cause_gestion_gare,prct_cause_prise_en_charge_voyageurs
0,2018-01,National,BORDEAUX ST JEAN,PARIS MONTPARNASSE,141,870,5,NaN,289,11.247809,3.693179,NaN,147,28.436735,6.511118,NaN,110,6.511118,44,8,36.134454,31.092437,10.924370,15.966387,5.042017,0.840336
1,2018-01,National,LE MANS,PARIS MONTPARNASSE,56,406,1,NaN,213,8.479969,4.567119,NaN,105,18.049048,5.363539,"Ce mois-ci, l'OD a été touchée par les inciden...",32,5.363539,9,4,20.000000,35.000000,16.666667,16.666667,8.333333,3.333333
2,2018-01,National,PARIS MONTPARNASSE,LA ROCHELLE VILLE,166,226,0,NaN,21,6.239683,0.286283,NaN,19,24.736842,2.938053,NaN,11,2.938053,6,1,22.222222,27.777778,16.666667,16.666667,5.555556,11.111111
3,2018-01,National,PARIS MONTPARNASSE,NANTES,124,508,3,NaN,71,7.235211,0.734290,NaN,58,33.726437,5.292211,NaN,39,5.292211,18,8,33.333333,22.222222,16.666667,20.370370,5.555556,1.851852
4,2018-01,National,POITIERS,PARIS MONTPARNASSE,94,472,4,NaN,224,6.784673,3.229701,NaN,89,14.592697,4.882372,NaN,42,4.882372,10,0,15.789474,45.614035,19.298246,15.789474,1.754386,1.754386


In [3]:
print(df.isna().sum())
print("Doublons :", df.duplicated().sum())

date                                        0
service                                     0
gare_depart                                 0
gare_arrivee                                0
duree_moyenne                               0
nb_train_prevu                              0
nb_annulation                               0
commentaire_annulation                  12544
nb_train_depart_retard                      0
retard_moyen_depart                         0
retard_moyen_tous_trains_depart             0
commentaire_retards_depart              12544
nb_train_retard_arrivee                     0
retard_moyen_arrivee                        0
retard_moyen_tous_trains_arrivee            0
commentaires_retard_arrivee             11846
nb_train_retard_sup_15                      0
retard_moyen_trains_retard_sup15            0
nb_train_retard_sup_30                      0
nb_train_retard_sup_60                      0
prct_cause_externe                          0
prct_cause_infra                  

In [4]:
df.describe().T

,count,mean,std,min,25%,50%,75%,max
duree_moyenne,12544.0,170.628508,87.730733,0.000000,99.000000,163.000000,223.000000,786.000000
nb_train_prevu,12544.0,272.623884,183.559230,0.000000,151.000000,230.000000,362.000000,1100.000000
nb_annulation,12544.0,8.407685,22.089205,0.000000,0.000000,2.000000,7.000000,297.000000
commentaire_annulation,0.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN
nb_train_depart_retard,12544.0,86.219866,88.428503,0.000000,22.000000,52.000000,126.000000,596.000000
retard_moyen_depart,12544.0,12.499854,11.660090,0.000000,6.346596,10.560933,16.101225,316.188095
retard_moyen_tous_trains_depart,12544.0,3.175135,5.054562,-229.269444,1.226076,2.368000,4.031985,115.047390
commentaire_retards_depart,0.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN
nb_train_retard_arrivee,12544.0,38.243304,31.826539,0.000000,16.000000,30.000000,52.000000,376.000000
retard_moyen_arrivee,12544.0,35.471909,15.605191,-40.109259,25.958062,33.777423,42.776681,299.600000


In [5]:
num = df.select_dtypes("number")
print("Valeurs négatives :\n", (num < 0).sum()[(num < 0).sum() > 0])

print("Trains prévus = 0 :", (df.nb_train_prevu == 0).sum())
print("Annulés > prévus :", (df.nb_annulation > df.nb_train_prevu).sum())
print("Durée trajet = 0 :", (df.duree_moyenne == 0).sum())

causes = [c for c in df.columns if c.startswith("prct_")]
print("Somme des causes ≠ 100 :", (df[causes].sum(axis=1).round() != 100).sum())

Valeurs négatives :
 retard_moyen_tous_trains_depart     170
retard_moyen_arrivee                  2
retard_moyen_tous_trains_arrivee    121
retard_moyen_trains_retard_sup15      8
nb_train_retard_sup_30               43
dtype: int64
Trains prévus = 0 : 73
Annulés > prévus : 63
Durée trajet = 0 : 73
Somme des causes ≠ 100 : 275


In [6]:
print("Gares :", df.gare_depart.nunique())
print("Liaisons :", (df.gare_depart + " → " + df.gare_arrivee).nunique())

annee = df.groupby(df.date.str[:4])[["nb_train_retard_arrivee", "nb_train_prevu", "nb_annulation"]].sum()
annee["ponctualite"] = 100 - annee.nb_train_retard_arrivee / (annee.nb_train_prevu - annee.nb_annulation) * 100
annee["ponctualite"].round(1)

Gares : 59
Liaisons : 130


date
2018    82.2
2019    86.2
2020    87.6
2021    88.7
2022    85.7
2023    85.3
2024    86.3
2025    84.9
2026    82.1
Name: ponctualite, dtype: float64

## Constats
- 12 544 lignes, 130 liaisons, 59 gares, janv. 2018 → juin 2026
- `date` en texte → à convertir
- `commentaire_annulation` et `commentaire_retards_depart` vides → à supprimer
- 73 lignes sans train prévu, 63 avec annulés > prévus → à supprimer
- Valeurs négatives impossibles (`nb_train_retard_sup_30`, retards moyens à −472 min) → à corriger
- Ponctualité à l'arrivée : 82,2 % (2018) → pic 88,7 % (2021) → 82,1 % (2026, 6 mois)